source/common/common/posix/thread_impl.cc

```diff
@@ -93,7 +93,7 @@ class ThreadImplPosix : public Thread {
   }
 
 private:
-#ifdef __linux__
+#ifdef SUPPORTS_PTHREAD_GETNAME_NP
```

Comment:
You probably want to cover the setname location in here, too.

Reply:
What's the symbol for this? I'm not seeing it appear in the bionic docs: https://android.googlesource.com/platform/bionic/+/master/docs/status.md

Reply:
`pthread_setname_np`

It's in this file as well.

Reply:
L55/62 :)

Reply:
It might be available without the API restriction (again, I'm assuming android bionicis law)

Reply:
With the logic in here though, it looks like usage will still be broken if you don't guard both locations.

Reply:
`getNameFromOS` is defined here and used there.

Reply:
Ah, that's what you mean. You're right the method `getNameFromOS()` wont be available on L55-62

Reply:
Yeah, it looks to me like both blocks currently guarded by `__linux__` should be included together or removed together, based on the environment.